aqueduct_client/aqueduct_client.py

```diff
@@ -0,0 +1,243 @@
+import datetime
+
+import requests
+import pandas as pd
+
+from .utils import load_api_key
+
+
+class AqueductClient(object):
+    def __init__(self, api_key=None, base_url=None):
+        """
+        Creates an AqueductClient.
+
+        Parameters
+        ----------
+        api_key : str, optional
+            The Quantopian API key to use.  If not given, we attempt
+            to load the key from a credentials file or from an
+            environment variable.
+
+        base_url : str, optional
+            The base URL for the Aqueduct API.  Defaults to the
+            FactSet Aqueduct endpoint.
+        """
+        if base_url:
+            self._base_url = base_url
+        else:
+            self._base_url = \
+                "https://factset.quantopian.com/api/experimental/pipelines"
+
+        if api_key is not None:
+            self._api_key = api_key
+        else:
+            self._api_key = load_api_key()
+
+    def get_all_pipelines(self):
+        """
+        Returns the metadata of all the pipelines you've run.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        List
+            A list of all the pipelines you have run.  Each pipeline is a dict
+            with id, start_date, end_date, created_at, status, and other
+            properties.
+        """
+        response = self._get('', None)
+        response.raise_for_status()
+        pipelines = response.json()['pipelines']
+        return pipelines
+
+    def get_pipeline(self, id):
+        """
+        Returns the metadata of a single pipeline.
+
+        Parameters
+        ----------
+        id : str
+            The id of the pipeline to load.
+
+        Returns
+        -------
+        dict
+            The metadata of the pipeline, containing id, start_date,
+            end_date, created_at, status, and other properties.
+        """
+        response = self._get('/{id}'.format(id=id), None)
+        response.raise_for_status()
+        pipeline = response.json()['pipeline']
```

Comment:
We might want to consider parsing any dates passed here back into proper date-like objects instead of strings. One interesting question that raises is what value we should return. On Q, we generally use `pd.Timestamp` with a timestamp of midnight UTC for representing dates, but I don't really like that convention. 

My first instinct would probably be to return `datetime.date` values for `start_date` and `end_date`, and `datetime.datetime` values for `created_at`. Since we're returning `pd.DataFrame` objects elsewhere though, I could see an argument for `pd.Timestamp` values for the datetimes, at which point it starts to feel odd that we're sometimes returning `date` and other times returning `Timestamp`, at which point we end up back at the midnight UTC convention... :/. I really wish pandas just had a `Date` class.